packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -1206,6 +1206,171 @@ describe('StripeHelper', () => {
     });
   });
 
+  describe('retrieveCouponDetails', () => {
+    it('retrieves coupon details', async () => {
+      const expected = {
+        promotionCode: 'promo',
+        type: 'forever',
+        discountAmount: 1,
+        valid: true,
+      };
+      sandbox.stub(stripeHelper, 'previewInvoice').resolves({
+        discount: {},
+        total_discount_amounts: [{ amount: 1 }],
+      });
+
+      sandbox.stub(stripeHelper, 'getCoupon').resolves({
+        duration: 'forever',
+        valid: true,
+      });
+
+      sandbox.stub(stripeHelper, 'retrievePromotionCodeForPlan').resolves({
+        active: true,
+        coupon: {
+          id: 'promo',
+        },
+      });
+
+      const actual = await stripeHelper.retrieveCouponDetails({
+        county: 'US',
+        planId: 'planId',
+        promotionCode: 'promo',
+      });
+      assert.deepEqual(actual, expected);
```

Comment:
I think we should assert the calls of the functions you stub'd above.  Otherwise we wouldn't know that `retrieveCouponDetails` called them and with what args.

Reply:
added, not sure why its not showing up here.